webview-ui/src/utils/hooks.ts

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useRef, useLayoutEffect, useState, useEffect } from "react"
+
+export const useShortcut = (shortcut: string, callback: any, options = { disableTextInputs: true }) => {
+	const callbackRef = useRef(callback)
+	const [keyCombo, setKeyCombo] = useState<string[]>([])
+
+	useLayoutEffect(() => {
+		callbackRef.current = callback
+	})
+
+	const handleKeyDown = useCallback(
+		(event: KeyboardEvent) => {
+			const isTextInput =
+				event.target instanceof HTMLTextAreaElement ||
+				(event.target instanceof HTMLInputElement && (!event.target.type || event.target.type === "text")) ||
+				(event.target as HTMLElement).isContentEditable
+
+			const modifierMap: { [key: string]: boolean } = {
+				Control: event.ctrlKey,
+				Alt: event.altKey,
+				Command: event.metaKey,
```

Comment:
Command is specific to OSX.  We really should use Meta here, as on linux, it's `Meta` on Windows it's `Win`, but internally to all three it's considered to be `Meta` key.